src/toasts/active-toasts.component.js

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import { toastsSubject } from "./toasts";
+import Toast from "./toast.component";
+
+export default function ActiveToasts() {
+  const [toasts, setToasts] = React.useState([]);
+  const [toastsClosing, setToastsClosing] = React.useState([]);
+  const closeToastRef = React.useRef();
+  closeToastRef.current = closeToast;
+
+  React.useEffect(() => {
+    const subscription = toastsSubject.subscribe(toast =>
+      setToasts([...toasts, toast])
+    );
+
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, [toastsSubject, toasts]);
+
+  React.useEffect(() => {
+    if (toastsClosing.length > 0) {
+      const timeoutId = setTimeout(() => {
```

Comment:
We're using CSS Transitions to animate the toast leaving the screen. What this means is we need to leave the HTML elements in the DOM during the transition, but then remove them afterwards.

Reply:
curious to know how a react component would work across frameworks?

Reply:
See https://github.com/openmrs/openmrs-esm-styleguide/pull/44#discussion_r331117907